CDCgov/prime-reportstream
# 14643 hl7 to fhir mappings validation

This PR adds validation for HL7 to FHIR mappings. It includes both messages and resources.

Test Steps:
1. See that CI passes
2. ```./gradlew primeCLI --args='validate-yaml --type HL7ToFhirMappingMessageTemplate --dir metadata/HL7/catchall/hl7/message metadata/HL7/v251-elr/hl7/message'```
3. ```./gradlew primeCLI --args='validate-yaml --type HL7ToFhirMappingResourceTemplate --dir metadata/HL7 --exclude-dir metadata/HL7/catchall/hl7/message metadata/HL7/v251-elr/hl7/message metadata/HL7/v251-elr/hl7/codesystem metadata/HL7/catchall/hl7/codesystem --exclude-file metadata/HL7/catchall/fhir/resourcemapping.yml metadata/HL7/v251-elr/fhir/resourcemapping.yml metadata/HL7/catchall/hl7/resource/Common.yml metadata/HL7/v251-elr/hl7/resource/Common.yml'```

## Changes
- Adds validation pipeline for HL7 to FHIR mappings

### Testing

### Merge first!
https://github.com/CDCgov/hl7v2-fhir-converter/pull/15
